src/Core/Checker.cpp

```diff
@@ -170,29 +171,36 @@ void Checker::onRunFinished(int index, const QString &, const QString &err, int
     if (tle)
         log->warn(tr("Checker[%1]").arg(index + 1), tr("Time Limit Exceeded"));
 
-    if (exitCode == 0)
+    switch (TResult(exitCode))
     {
-        // the check process succeeded
+    case _ok:
         if (!err.isEmpty())
             log->message(tr("Checker[%1]").arg(index + 1), err, "green");
         emit checkFinished(index, Widgets::TestCase::AC);
-    }
-    else if (QList<int>({1, 2, 3, 4, 5, 8, 16}).contains(exitCode)) // this list is from testlib.h::TResult
-    {
-        // This exit code is a normal exit code of a testlib checker, means WA or something else
+        return;
+
+    case _wa:
+    case _pe:
+    case _fail:
+    case _dirt:
+    case _points:
+    case _unexpected_eof:
+    case _partially:
```

Comment:
It would be great if for each special WA, the checker could report some reason. Something like `Checker[3]: Wrong answer, Unexpected End of file, output has less contents than expected`.

Nothing special should be displayed if WA is because of output/expected mismatch.

Reply:
That's in the stderr of the checker.

Reply:
If checker returns `_unexpected_eof` we can simply `emit(index, WA)` we could `log->info("Checker [%1]".arg(index), "Early EOF encountered")` before we emit WA.

In this way, the user has some more information that WA because of Early EOF.

Reply:
As I said, the more detailed information is in the stderr of the checker.

Reply:
Is stderr of checker logged in the Messagelogger?

Reply:
> Is stderr of checker logged in the Messagelogger?

Yes. You'd better read the codes or have a try. I said twice it's in the stderr, which means it's in the message logger.